### PR TITLE
Rust guessing game

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: vm/examples/guessing-game/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+fn init() {
+    // Set seed to some random value, just to show how to use init()
+    SEED.with(|seed| *seed.borrow_mut() = 123456789);
+}
+
+#[invocation_handler(init_fn = init)]
+fn game(input: String) -> String {
+    count_request(&input);
+    update_seed(&input);
+    init_secret();
```
Comment (michaelvoronov) [inline vm/examples/guessing-game/src/lib.rs]: Hmm, seems like a good candidate to move to `init` function. And in this case `update_secret` could be use instead.

File: vm/examples/guessing-game/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                next_game();
+                format!("Success! You guessed right.")
+            }
+        },
+    }
+}
+
+/// Increment number of tries
+fn count_request(input: &String) {
+    TRIES.with(|t| *t.borrow_mut() += 1);
```
Comment (michaelvoronov) [inline vm/examples/guessing-game/src/lib.rs]: It is also possible to use `.add_assign(1)` here (`std::ops::AddAssign` should be included). But it is up to you.

Comment (folex) [reply vm/examples/guessing-game/src/lib.rs]: It would still look like
```rust
TRIES.with(|t| t.borrow_mut().add_assign(1))
```

so even more verbose

Comment (michaelvoronov) [reply vm/examples/guessing-game/src/lib.rs]: I don't think so. For me using `*` is less clear that `add_assign`. Btw it is very minor and both variant are good (e.g. in `ttt` i used `add_assign`).

**Pull request merged**